src/Sparrow/Json/JsonOperationContext.cs

```diff
@@ -520,10 +524,10 @@ public virtual void Reset()
 
 
             foreach (var disposable in _disposables)
-            {
                 disposable.Dispose();
-            }
+
             _disposables.Clear();
+            _lastStreamSize = InitialStreamSize; // this must be done last, because _disposables can change this value
```

Comment:
The point here is that the next time that we need to send a context out, we'll already allocate it at the required size.
Admittedly, now that we are using arena allocator it isn't as important, but it avoid having to copy the data after we created the doc.
Why do you reset it like this?